sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/SymbolicExecution/ISymbolicExecutionAnalysisContext.cs

```diff
@@ -28,6 +28,8 @@ namespace SonarAnalyzer.Rules.SymbolicExecution
     // or cached nodes) and clear it and the end.
     public interface ISymbolicExecutionAnalysisContext : IDisposable
     {
+        bool SupportPartialWalk { get; }
```

Comment:
nitpick: I would use `SupportsPartialWalking` (it's a state rather than an action)

Also - can we find a better name or add a comment on the interface - what does "support partial walking mean"?

I need to look in `SymbolicExecutionRunner.cs` to tell, and it's not very clear - for those rules that support partial walking, I raise _only_ during the run and not after the full exploration, whereas for the other ones, which don't support Partial Walking, I run _only_ when the exploration ended.

So from what I see, `SupportPartialWalking` is exclusive with `ReportAtExplorationEnd`.

Whereas when I first read "Support Partial Walking", I imagined this is an additive / complementary attribute, like "I do what the other rules do AND I also do stuff with partial results".

I think we need somewhere to document this behavior.

Reply:
WDYT about `SupportPartialResults`? Since it's not the context responsibility to know about the tree walking/exploration.